### ISSUE TITLE
Detach on DTUM forward pass

### ISSUE THREAD

**Issue by ninte** [TinaLRJ/Multi-frame-infrared-small-target-detection-DTUM#8]
Hi, checking the code I've found that you are performing a detach on o_2 at the forward pass of DTUM:
https://github.com/TinaLRJ/Multi-frame-infrared-small-target-detection-DTUM/blob/main/models/layers.py

        o_3 = self.relu(self.bn3_2(self.conv3_2(torch.cat([self.up(x_4),x_3], dim=1))))
        o_2 = self.relu(self.bn2_2(self.conv2_2(torch.cat([self.up(o_3),x_2], dim=1)))).detach()
        o_1 = self.relu(self.bn1_2(self.conv1_2(torch.cat([self.up(o_2),x_1], dim=1))))

        x_out = self.final(torch.cat([o_1, torch.unsqueeze(x[:,:,-1,:,:],2)], dim=1))

        return x_out

What is the reason to do this?

Thanks :D

**Comment by MrNeoBlue** [comment]
I'm also confused about this line of code — does detach affect the data flow?

**Comment by TinaLRJ** [comment]
Removing the .detach() caused training errors in my environment (Python 3.7 and CUDA 11.7), but retaining it enabled stable training and achieved SOTA performance. This suggests that newer environment versions might allow .detach() removal without issues.

**Comment by TinaLRJ** [comment]
I'm also keen to understand the root cause of errors when omitting detach()! Feel free to share your insights here!
